server/src/main/java/org/elasticsearch/cluster/routing/RoutingTable.java

```diff
@@ -242,6 +242,10 @@ public ShardsIterator allShards(String[] indices) {
         return allShardsSatisfyingPredicate(indices, Predicates.always(), false);
     }
 
+    public ShardsIterator allSearchableShards(String[] indices) {
```

Comment:
Perhaps javadoc could be helpful here to describe exactly what "searchable" means (i.e., on a node having a role that makes it able to be targeted for searches)